src/test_workflow/dependency_installer.py

```diff
@@ -52,17 +53,19 @@ def download_dist(self, dest):
 
     def download(self, paths, category, dest):
         logging.info(f"Downloading to {dest} ...")
-        for path in paths:
-            url = "/".join([self.root_url, category, path])
-            # paths are prefixed by category, remove
-            local_path = os.path.join(dest, "/".join(path.split("/")[1:]))
-            self.__download_or_copy(url, local_path)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
```

Comment:
Nit: `max_workers` should default to number of CPUs.

Reply:
Download/Copy isn't CPU bound (typically) I want to make sure docker image copy is accelerated as well.